library/Freckle/App/Dotenv.hs

```diff
@@ -0,0 +1,56 @@
+-- | Conventional @.env@ handling
+module Freckle.App.Dotenv
+  ( load
+  , loadTest
+  , loadFrom
+  ) where
+
+import Freckle.App.Prelude
+
+import qualified Configuration.Dotenv as Dotenv
+import GHC.Stack (HasCallStack)
+import System.FilePath (takeDirectory, (</>))
+import UnliftIO.Directory
+  (doesFileExist, getCurrentDirectory, withCurrentDirectory)
+import UnliftIO.Exception (throwString)
+
+-- | Call 'loadFrom' with @.env@
+load :: HasCallStack => IO ()
+load = loadFrom ".env"
+
+-- | Call 'loadFrom' with @.env.test@
+loadTest :: HasCallStack => IO ()
+loadTest = loadFrom ".env.test"
+
+-- | Load the given file, asserting it contains all variables in @.env.example@
```

Comment:
Should add "...when present".

I decided to support a lack of `.env.example` (for now), just to ease transition of existing apps.